pkg/internal/config/config_test.go

```diff
@@ -144,6 +144,18 @@ func TestNewShouldFailWhen(t *testing.T) {
 			faultyKeyValue: "",
 			expectedError:  config.ErrInvalidNetworkAttachmentDefinitionName,
 		},
+		{
+			description:    "trafficGeneratorNodeLabelSelector is missing and DPDKNodeLabelSelector is set",
+			key:            config.TrafficGeneratorNodeLabelSelectorParamName,
+			faultyKeyValue: "",
+			expectedError:  config.ErrIllegalLabelSelectorCombination,
+		},
+		{
+			description:    "DPDKNodeLabelSelector is missing and trafficGeneratorNodeLabelSelector is set",
+			key:            config.DPDKNodeLabelSelectorParamName,
+			faultyKeyValue: "",
+			expectedError:  config.ErrIllegalLabelSelectorCombination,
+		},
```

Comment:
Is there also a positive test that checks the validity of the config when both fields are set or both unset?

Reply:
`TestNewShouldApplyUserConfig` checks the positive scenario of explicitly setting the nodes.
Do you suggest I split it into 2 cases?

Reply:
> 

> `TestNewShouldApplyUserConfig` checks the positive scenario of explicitly setting the nodes. Do you suggest I split it into 2 cases?

Yes. I suggest to test the case where both fields are full and also when both are empty.

Reply:
DONE